src/core/sock/sock-extra.cpp

```diff
@@ -83,12 +83,7 @@ extern "C" int xlio_recvfrom_zcopy(int __fd, void *__buf, size_t __nbytes, int *
         *__flags |= MSG_XLIO_ZCOPY;
         return p_socket_object->rx(RX_RECVFROM, piov, 1, __flags, __from, __fromlen);
     }
-    BULLSEYE_EXCLUDE_BLOCK_START
-    if (!orig_os_api.recvfrom) {
```

Comment:
Should we ignore this check?

Reply:
Please referer https://github.com/alexbriskin/libxlio_public/blob/ac507222d23e75a0d0f09090eec186187a357ea3/src/core/sock/sock-redirect.h#L105

The answer is that the SYSCALL macro in the dynamic library case:

1. Checks the syscall initialized in the orig_os_api object
2. If not, initializes the object
3. Invoke the function pointer

In the static case, the SYSCALL macro calls the original symbols without the xlio_ prefix.